pygmt/tests/test_plot3d.py

```diff
@@ -0,0 +1,582 @@
+        B=["ag", "zag"],
+    )
+    fig_test.plot3d(
+        x=data[:, 0],
+        y=data[:, 1],
+        z=data[:, 2],
+        zscale=5,
+        perspective=[225, 30],
+        region=region,
+        projection="R270/4i",
+        style="s1c",
+        color="green",
+        frame=["ag", "zag"],
+    )
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_plot3d_colors(data, region):
+    "Plot the data using z as colors"
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.plot3d(
+        data=POINTS_DATA,
+        Jz=5,
+        p="225/30",
+        G="+z",
+        R="/".join(map(str, region)),
+        J="X3i",
+        S="c0.5c",
+        C="cubhelix",
+        B=["afg", "zafg"],
+        i="0,1,2,2",
+    )
+    fig_test.plot3d(
+        x=data[:, 0],
+        y=data[:, 1],
+        z=data[:, 2],
+        zscale=5,
+        perspective=[225, 30],
+        color=data[:, 2],
+        region=region,
+        projection="X3i",
+        style="c0.5c",
+        cmap="cubhelix",
+        frame=["afg", "zafg"],
+    )
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_plot3d_sizes(data, region):
+    "Plot the data using z as sizes"
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.plot3d(
+        data=POINTS_DATA,
+        Jz=5,
+        p="225/30",
+        i="0,1,2,2+s0.5",
+        R="/".join(map(str, region)),
+        J="X4i",
+        S="cc",
+        G="blue",
+        B=["af", "zaf"],
+    )
+    fig_test.plot3d(
+        x=data[:, 0],
+        y=data[:, 1],
+        z=data[:, 2],
+        zscale=5,
+        perspective=[225, 30],
+        sizes=0.5 * data[:, 2],
+        region=region,
+        projection="X4i",
+        style="cc",
+        color="blue",
+        frame=["af", "zaf"],
+    )
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_plot3d_colors_sizes(data, region):
+    "Plot the data using z as sizes and colors"
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.plot3d(
+        data=POINTS_DATA,
+        Jz=5,
+        p="225/30",
+        i="0,1,2,2,2+s0.5",
+        R="/".join(map(str, region)),
+        J="X3i",
+        S="cc",
+        C="copper",
+        B=["af", "zaf"],
+    )
+    fig_test.plot3d(
+        x=data[:, 0],
+        y=data[:, 1],
+        z=data[:, 2],
+        zscale=5,
+        perspective=[225, 30],
+        color=data[:, 2],
+        sizes=0.5 * data[:, 2],
+        region=region,
+        projection="X3i",
+        style="cc",
+        cmap="copper",
+        frame=["af", "zaf"],
+    )
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_plot3d_colors_sizes_proj(data, region):
+    "Plot the data using z as sizes and colors with a projection"
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.plot3d(
+        data=POINTS_DATA,
+        Jz=5,
+        p="225/30",
+        R="/".join(map(str, region)),
+        J="M10i",
+        B=["af", "zaf"],
+        G="+z",
+        i="0,1,2,2,2+s1",
```

Comment:
Ok, been trying (unsuccessfully) to see why 3 Windows tests are failing. Specifically `test_plot3d_sizes`, `test_plot3d_colors_sizes`, `test_plot3d_colors_sizes_proj`. I can't get my university Windows box to run `pytest` somehow today (keep getting a `pygmt.exceptions.GMTCLibError: Module 'psconvert' failed with status code 78:
E           psconvert [ERROR]: System call [@gswin64c -q ... returned error 1]` message). So no idea what the image difference looks like other than a vague GMTImageComparisonFailure message.

```
E               pygmt.exceptions.GMTImageComparisonFailure: images not close (RMS 28.107):
E               	result_images\test_plot3d_colors_sizes_proj[png].png
E               	result_images\test_plot3d_colors_sizes_proj[png]-expected.png
```


**However**, I think it might have to do with using the column scaling `i=2+s1`, since the 3 test failures have these. Is this something Windows is particularly sensitive to?

Reply:
We should upload the testing directory (`tmp-test-dir-with-unique-name`) as an artifact (https://github.com/actions/upload-artifact) if the testing fails.

Reply:
Yep, it's to do with the `i="2+s1"` scaling problem. I can try to refactor the test a bit to make it work cross-platform, but this bug might hit some of our users at a later date.

| expected  | actual  | diff  |
|---|---|---|
| ![test_plot3d_colors_sizes_proj png](https://user-images.githubusercontent.com/23487320/97254640-b2807200-1873-11eb-996f-43d08cbece8c.png) | ![test_plot3d_colors_sizes_proj png -expected](https://user-images.githubusercontent.com/23487320/97254644-b3b19f00-1873-11eb-8fcf-26a69f06afb0.png)| ![test_plot3d_colors_sizes_proj png -failed-diff](https://user-images.githubusercontent.com/23487320/97254646-b3b19f00-1873-11eb-8d82-8c887af89bba.png) |

P.S. Will cherry-pick commit 413ff77c98facb8370cd5bf8e53c6dfe54bd7848 into a separate Pull Request.

Reply:
> P.S. Will cherry-pick commit [413ff77](https://github.com/GenericMappingTools/pygmt/commit/413ff77c98facb8370cd5bf8e53c6dfe54bd7848) into a separate Pull Request.

There are 3 failing jobs, but it seems I only see one artifact. Perhaps in 413ff77c98facb8370cd5bf8e53c6dfe54bd7848 we should use unique names for artifacts from different jobs?

Reply:
> There are 3 failing jobs, but it seems I only see one artifact. Perhaps in [413ff77](https://github.com/GenericMappingTools/pygmt/commit/413ff77c98facb8370cd5bf8e53c6dfe54bd7848) we should use unique names for artifacts from different jobs?

Isn't it a separate artifact under each job (i.e. WIndows Py36/Py37/Py38?), I didn't look too closely at the files but I assumed it would be separate.

Reply:
I only see one artifact in this page: https://github.com/GenericMappingTools/pygmt/actions/runs/330476822

Reply:
Anyways, I think there might be a bug (or feature?) here somewhere, as using `i="0,1,2,2"` with `-Suc` doesn't change the unit to **c**entimetres, but keeps on using inches as the default unit. The documentation at https://docs.generic-mapping-tools.org/latest/plot3d.html#s confuses me, especially with the mention of `PROJ_LENGTH_UNIT`.

```
gmt begin temp png
  gmt plot3d pygmt/tests/data/points.txt -Jz5 -p225/30 -i0,1,2,2 -R10/70/-5/10/0/1 -JX4i -Suc -Gblue -Baf -Bzaf
gmt end show
```

produces cubes sized in inches instead of centimetres

![temp](https://user-images.githubusercontent.com/23487320/97256549-2a509b80-1878-11eb-82b0-06f3128d9b78.png)

Reply:
I think you're right. **-Suc** and **-Sui** give the same result. It may be a GMT bug.

Reply:
Great, time to open up a new GMT bug report then. I've also made a new PR on uploading artifacts opened at #675 so discussion can continue there.

In the meantime, I'll change these tests to use inches which will hopefully make things pass on Windows.

Reply:
Please report the bug to GMT, and also add a few comments in the tests explaining why we use inches.

Reply:
Cool, all tests green on commit 878a07351afbf17928e0293dfb532464caf1e94a when using inches, even with column scaling `i="2+s0.5". I'll revert 413ff77c98facb8370cd5bf8e53c6dfe54bd7848 in this PR (since #675 is opened) and re-request a review to trigger the GMT Latest tests.

Reply:
> Please report the bug to GMT, and also add a few comments in the tests explaining why we use inches.

Ok, bug reported at https://github.com/GenericMappingTools/gmt/issues/4386, and I've added notes to the 3 unit tests on why inches are used instead of centimetres at commit 256a0a9b1a9333827ec9a9e8b36e7b45785d4a19.